diff --git a/src/main.cpp b/src/main.cpp
index e975a6b9..19c86f7b 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -59,6 +59,10 @@ int main(int argc, char** argv)
     }
 #endif
 
+    // snap: force fallback theme to avoid using system theme (gtk integration) with missing actions
+    // just like on Windows and macOS
+    QIcon::setThemeSearchPaths(QStringList() << ":/icons");
+
     Application app(argc, argv);
     Application::setApplicationName("keepassxc");
     Application::setApplicationVersion(KEEPASSXC_VERSION);
